backend/base/tests.py

```diff
@@ -1,3 +1,98 @@
-from django.test import TestCase
+from base.serializers import UserSerializerWithToken
+from django.contrib.auth.models import User
+from rest_framework import status
+from rest_framework.test import APITestCase
 
 # Create your tests here.
+
+
+class RegistrationTestCase(APITestCase):
+    def test_registration(self):
+        data = {
+            "name": "testcase",
+            "email": "test@localhost.app",
+            "password": "some_psw",
+        }
+
+        response = self.client.post("/api/users/register/", data)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+
+class UserProfileTestCase(APITestCase):
+    """Handle user profile (by a user)"""
+
+    def setUp(self):
+        self.user = User.objects.create(
+            first_name="testcase",
+            username="test@localhost.app",
+            email="test@localhost.app",
+            password="some_psw",
+        )
+        serializer = UserSerializerWithToken(self.user, many=False)
+        self.api_authentication(serializer.get_token(self.user))
+
+    def api_authentication(self, token):
+        self.client.credentials(HTTP_AUTHORIZATION="Bearer " + token)
+
+    def test_profile_authenticated(self):
+        response = self.client.get(
+            "/api/users/profile/", content_type="application/json"
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_profile_not_authenticated(self):
+        self.client.force_authenticate(user=None)
+        response = self.client.get(
+            "/api/users/profile/", content_type="application/json"
+        )
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_profile_update_user(self):
+        # response = self.client.put(reverse('profile-detail', kwargs={'pk': 1}))
+        response = self.client.put(
+            "/api/users/update_profile/",
+            data={
+                "id": self.user.id,
+                "name": "testcase2",
+                "email": "test2@localhost.app",
+                "password": "strong-pswd",
+            },
+            content_type="application/json",
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```

Comment:
@citos88 zwraca mi bad request, problem jest z danymi, ponieważ w widoku przy komendzie data = request.data zwraca błąd. Nie wiem dlaczego, skoro input daję dokładnie taki sam jak front.

Reply:
spróbuj self.client.put(url, data, format='json')

Reply:
dziękuję bardzo, zadziałało jak magia, co to znaczy oko profesjonalisty 🙂